### PR TITLE
Fix Docker Executable with optional Executable, PHP-CS-Fixer

### PR DESCRIPTION
**Pull request by Glavin001**
### What does this implement/fix? Explain your changes.

- Optional Executables do not attempt to use Docker when not installed.
- PHP-CS-Fixer does not work with Docker Executable

### Does this close any currently open issues?

No, I do not think so.

### Any other comments?

See #1707.

### Checklist

Check all those that are applicable and complete.

- [x] Merged with latest `master` branch
- [x] Regenerate documentation with `npm run docs`
- [x] Add change details to `CHANGELOG.md` under "Next" section
- [ ] Added examples for testing to [examples/ directory](examples/)
- [ ] Travis CI passes (Mac support)
- [ ] AppVeyor passes (Windows support)

### CODE REVIEW

File: src/beautifiers/executable.coffee [hunk 1]
```diff
@@ -49,6 +49,7 @@ class Executable
       .then(() => @)
       .catch((error) =>
         if not @.required
+          @verbose("Not required")
```
Comment (Glavin001) [inline src/beautifiers/executable.coffee]: Add executable name.

File: src/beautifiers/executable.coffee [hunk 2]
```diff
@@ -366,6 +367,7 @@ class HybridExecutable extends Executable
 
   constructor: (options) ->
     super(options)
+    @verbose("HybridExecutable Options", options)
```
Comment (Glavin001) [inline src/beautifiers/executable.coffee]: Add executable name.

File: src/beautifiers/executable.coffee (hunk 1 above)
Comment (Glavin001) [reply src/beautifiers/executable.coffee]: Nevermind.

image

File: src/beautifiers/executable.coffee (hunk 2 above)
Comment (Glavin001) [reply src/beautifiers/executable.coffee]: Nevermind.

image

**Comment by szeck87** [comment]
Travis failed for an unrelated issue.  Merging.

**Pull request merged**